src/cmd/version.go

```diff
@@ -1,24 +1,104 @@
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
-var version = "development"
+type Build struct {
+	Version         string          `json:"version,omitempty"`
+	Commit          string          `json:"git,omitempty"`
+	GoInfo          GoInfo          `json:"go,omitempty"`
+	OpslevelVersion OpslevelVersion `json:"opslevel,omitempty"`
+}
+
+type OpslevelVersion struct {
+	Commit  string `json:"app_commit,omitempty"`
+	Version string `json:"app_version,omitempty"`
+}
+
+type GoInfo struct {
+	Version  string `json:"version,omitempty"`
+	Compiler string `json:"compiler,omitempty"`
+	OS       string `json:"os,omitempty"`
+	Arch     string `json:"arch,omitempty"`
+}
+
+var (
+	version = "development"
+	commit  = "none"
+	build   Build
+)
+
+func initBuild() {
+	build.Version = version
+	if len(commit) >= 12 {
+		build.Commit = commit[:12]
+	} else {
+		build.Commit = commit
+	}
+
+	build.GoInfo = getGoInfo()
+	build.OpslevelVersion = getOpslevelVersion()
+}
+
+func getGoInfo() GoInfo {
+	return GoInfo{
+		Version:  runtime.Version(),
+		Compiler: runtime.Compiler,
+		OS:       runtime.GOOS,
+		Arch:     runtime.GOARCH,
+	}
+}
+
+func getOpslevelVersion() OpslevelVersion {
+	// Need to update all of this when we switch over to resty client
+	apiUrl := viper.GetString("base-url") + "/api/ping"
```

Comment:
Tried to get path.Join to work here but was getting this:  ```Get "https:///app.opslevel-staging.com/api/ping": http: no Host in request URLexit status 1```
Looked it up but not sure why it was adding an extra slash after the protocol.

Reply:
Ya we need to sanitize this because our end user will be setting `base-url` and they might add a / at the end

lets do this
```
url := url.Parse(viper.GetString("base-url"))
url.Path = "/api/ping"
response, err := http.Get(url.String())
````

Reply:
Oh nice, yeah that's better.  Just updated it.